10_rest/mohitnag/pkg/rest/rest_test.go

```diff
@@ -0,0 +1,251 @@
+
+func (s *restSuite) TestHandleUpdatePuppy() {
+	puppyServer := httptest.NewServer(s.handler)
+	assert := assert.New(s.T())
+	update := puppy.Puppy{
+		ID:     1,
+		Breed:  "dog",
+		Colour: "red",
+		Value:  "20",
+	}
+	b, err := json.Marshal(update)
+	assert.NoError(err)
+	req, err := http.NewRequest(http.MethodPut, puppyServer.URL+"/api/puppy", bytes.NewBuffer(b))
+	assert.NoError(err)
+	req.Header.Set("Content-Type", "application/json; charset=utf-8")
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	assert.NoError(err)
+	defer resp.Body.Close()
+	actual, err := ioutil.ReadAll(resp.Body)
+	assert.NoError(err)
+	updatedPuppy := puppy.Puppy{}
+	err = json.Unmarshal(actual, &updatedPuppy)
+	assert.NoError(err)
+	assert.Equal("red", updatedPuppy.Colour)
+}
+
+func (s *restSuite) TestHandleUpdateMissingPuppy() {
+	assert := assert.New(s.T())
+	puppyServer := httptest.NewServer(s.handler)
+	updatePuppy := puppy.Puppy{
+		ID:     100,
+		Breed:  "dog",
+		Colour: "red",
+		Value:  "2",
+	}
+	b, err := json.Marshal(updatePuppy)
+	assert.NoError(err)
+	req, err := http.NewRequest(http.MethodPut, puppyServer.URL+"/api/puppy", bytes.NewBuffer(b))
+	assert.NoError(err)
+	req.Header.Set("Content-Type", "application/json; charset=utf-8")
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	assert.NoError(err)
+	defer resp.Body.Close()
+	actual, err := ioutil.ReadAll(resp.Body)
+	assert.NoError(err)
+	updatedPuppy := puppy.Puppy{}
+	err = json.Unmarshal(actual, &updatedPuppy)
+	assert.NoError(err)
+	assert.Equal(404, resp.StatusCode)
+	assert.Equal(puppy.Puppy{}, updatedPuppy)
+}
+
+func (s *restSuite) TestHandleUpdateBadPuppy() {
+	assert := assert.New(s.T())
+	puppyServer := httptest.NewServer(s.handler)
+	b, err := json.Marshal(`{bad json}`)
+	assert.NoError(err)
+	req, err := http.NewRequest(http.MethodPut, puppyServer.URL+"/api/puppy", bytes.NewBuffer(b))
+	assert.NoError(err)
+	req.Header.Set("Content-Type", "application/json; charset=utf-8")
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	assert.NoError(err)
+	defer resp.Body.Close()
+	actual, err := ioutil.ReadAll(resp.Body)
+	assert.NoError(err)
+	updatedPuppy := puppy.Puppy{}
+	err = json.Unmarshal(actual, &updatedPuppy)
+	assert.NoError(err)
+	assert.Equal(400, resp.StatusCode)
+	assert.Equal(puppy.Puppy{}, updatedPuppy)
+}
+
+func (s *restSuite) TestDeletePuppy() {
+	assert := assert.New(s.T())
+	puppyServer := httptest.NewServer(s.handler)
+	req, err := http.NewRequest(http.MethodDelete, puppyServer.URL+"/api/puppy/1", nil)
+	assert.NoError(err)
+	req.Header.Set("Content-Type", "application/json; charset=utf-8")
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	assert.NoError(err)
+	defer resp.Body.Close()
+	assert.Equal(http.StatusNoContent, resp.StatusCode)
+	resp, err = http.Get(puppyServer.URL + "/api/puppy/1")
+	assert.NoError(err)
+	defer resp.Body.Close()
+	assert.Equal(404, resp.StatusCode)
+}
+
+func (s *restSuite) TestHandleDeleteMissingPuppy() {
```

Comment:
I would call this `TestHandleDeleteBadID`, and when you write the next test, that should be called `TestHandleDeleteMissingPuppy`

Reply:
Done